src/diagram/diagram-api.tsx

```diff
@@ -4,68 +4,164 @@
  */
 export interface DiagramActions {
 
+
   // Groups
 
+  /**
+   * @returns The list of groups registered in diagram.
+   */
   getGroups(): Group[];
 
+
+  /**
+   * Registers new {@link group} to the diagram.
+   * @param group is the group to be registered.
+   * @param content is optional content of the group.
+   */
   addGroup(group: Group, content?: string[]): void;
 
+
+  /**
+   * Removes given groups from diagram.
+   * @param groups is list of identifiers of the to be removed groups.
+   */
   removeGroups(groups: string[]): void;
 
+
+  /**
+   * Sets the content of given {@link group} to given {@link content}.
+   * @param group is the group to set content for.
+   * @param content is the new content of the group.
+   */
   setGroup(group: Group, content: string[]): void;
 
+
+  /**
+   * The content of {@link group} as node identifiers.
+   * @param group is the group to get content for.
+   * @returns The content of group as node identifiers.
+   */
   getGroupContent(group: Group): string[];
 
+
   // Nodes
 
+  /**
+   * @returns The nodes registered inside diagram.
+   */
   getNodes(): Node[];
```

Comment:
When I was trying to implement some of the layouting, I tried to do it on top of the Visual Model, but visual model doesn't have information about the widths and heights. They are based on how reactflow renders them. So maybe it makes sense to add method to query width/height of a node.

For example for aligning to the right side of selection we need to find the most right point, which is the highest number of the following sum `node.position.x + node.width`.